compiler/internal/codegen/codegen_main.go

```diff
@@ -755,29 +760,25 @@ func (b *Builder) writeDecoder(f *File) {
 }
 
 func (b *Builder) typeName(param *est.Param, skipPtr bool) *Statement {
-	// Reconstruct a named type to make the schemaTypeToGoType function slightly easier to recursive over
-	// TODO(domblack): in the future we should change `est.Param` to contain the schema.Type directly
-	typName := b.schemaTypeToGoType(&schema.Type{Typ: &schema.Type_Named{Named: &schema.Named{
-		Id:            param.Decl.Id,
-		TypeArguments: param.TypeArguments,
-	}}})
+	typName := b.schemaTypeToGoType(param.Type)
 
 	if param.IsPtr && !skipPtr {
 		typName = Op("*").Add(typName)
 	}
+
 	return typName
 }
 
 func (b *Builder) authDataType() Code {
 	s := ""
+
 	if ah := b.res.App.AuthHandler; ah != nil && ah.AuthData != nil {
-		t := ah.AuthData
-		s = t.Decl.Loc.PkgPath + "." + t.Decl.Name
-		if t.IsPtr {
-			s = "*" + s
-		}
+		typ := b.typeName(ah.AuthData, false)                           // Get the type
+		statement := Var().Id("a").Add(typ)                             // place within a var statement to force `typ` to render as a type, rather than a statement
+		s = strings.TrimPrefix(fmt.Sprintf("%#v", statement), "var a ") // render the statement, trimming the prefix
```

Comment:
Without this `var a ` prefix, a byte slice would be rendered as `[] byte` and not `[]byte`